src/components/Nav.js

```diff
@@ -0,0 +1,38 @@
+import React, {PropTypes} from 'react'
+import {Link} from 'react-router'
+import ToggleButton from './ToggleButton'
+import {getClient} from '../services/contentfulClient'
+import styles from './Nav.css'
+import CSSModules from 'react-css-modules'
+
+class Nav extends React.Component {
+  constructor () {
+    super()
+    this.state = {currentTokenType: 'preview'}
+  }
+  handleChange (isPreview) {
+    this.setState({currentTokenType: isPreview ? 'preview' : 'production'})
+  }
+  getNav () {
+    if (getClient()) {
+      return (
+        <ul>
+          <li styleName={this.state.currentTokenType === 'preview' ? 'selected' : ''}>Preview</li>
+          <li><ToggleButton changeHandler={this.handleChange.bind(this)}/></li>
```

Comment:
I am not sure but I remember in this case I had an error trying to access the setState saying that this is null so I had to add the bind